graphlearn/src/include/data_source.h

```diff
@@ -83,6 +83,9 @@ struct NodeSource {
   IndexOption   option;
   bool          local_shared;
 
+  std::string view_type;
```

Comment:
Are these two args only used in vineyard ? If so, why is it added to each graph creation function?

Reply:
Yes, those two argument are only used by the vineyard implementation. But the dispatch happens inside the macro `CREATE(Type)` (see https://github.com/alibaba/graph-learn/blob/master/graphlearn/src/core/graph/storage_creator.cc#L21).

The macro was updated as follows when integrating with vineyard in 2020,

https://github.com/alibaba/graph-learn/blob/641b305d74ed93f82b692c76039999ec6a32e5e6/graphlearn/core/graph/storage_creator.cc#L21-L40

Thus we need to pass these two arguments util reaching the macro, even they are not used by other storage implementation.